clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/CompletableEventReaper.java

```diff
@@ -102,6 +102,8 @@ public void reap(long currentTimeMs) {
         // Second, remove any events that are already complete, just to make sure we don't hold references. This will
         // include any events that finished successfully as well as any events we just completed exceptionally above.
         tracked.removeIf(e -> e.future().isDone());
+        
+        return tracked;
```

Comment:
I'm not sure I understand this pattern. Can we add a new method that returns the tracked events instead of overloading the purpose of `reap()`?